LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,352 @@
+	const Type& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	//-----------------------------------------------
+	Type& GetTail()
+	{
+		return Tail->Element;
+	}
+
+	const Type& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	//---------------------------------Size
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	//------------------------------------------------
+	int GetSize()
+	{
+		return Size;
+	}
+
+	//-----------------------------------Modifiers
+	void AddHead(const Type& InHead)
+	{
+		FNode* NewHead = new FNode;
+		NewHead->Element = InHead;
+
+		if (Head == nullptr)
+		{
+			Head = NewHead;
+			Tail = NewHead;
+			Size++;
+		}
+		else
+		{
+			NewHead->Next = Head;
+			Head = NewHead;
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------
+	void AddTail(const Type& InTail)
+	{
+
+		if (Head == nullptr)
+		{
+			AddHead(InTail);
+		}
+
+		else
+		{
+			Tail->Next = new FNode;
+			Tail->Next->Element = InTail;
+			Tail = Tail->Next;
+			Size++;
+		}
+	}
+	
+	//----------------------------------------------------
+	void Insert(const Type& NewItem, const int Index)
+	{
+		if (Index == 0)
+		{
+			AddHead(NewItem);
+		}
+		else if(Index == Size)
+		{
+			AddTail(NewItem);
+		}
+
+		else if (Index > 0 && Index < Size)
+		{
+			FNode* Temporal = Head;
+			FNode* Container = new FNode;
+
+			for (int i = 0; i < Index; i++)
+			{
+				Temporal = Temporal->Next;
+			}
+			Container->Element = Temporal->Element;
+
+			Temporal->Element = NewItem;
+			Container->Next = Temporal->Next;
+			Temporal->Next = Container;
+			Size++;
+		}
+	}
+	//---------------------------------------------------------
+	void RemoveHead()
+	{
+		if (Size != 0)
+		{
+			FNode* NewHead = Head->Next;
+			delete Head;
+			Head = NewHead;
+			--Size;
+		}
+
+		else if (Size == 1)
+		{
+			delete Head;
+			Head = nullptr;
+			Tail = nullptr;
+			--Size;
+		}
+	}
+
+	//---------------------------------------------------------
+	void Remove(const int Index)
+	{
+		FNode* Temporal = Head;
+
+		if (Index >= 0 && Index <= Size)
+		{
+			for (int i = 0; i < Index - 1; i++)
+			{
+				Temporal = Temporal->Next;
+			}
+
+			FNode* Current = Temporal->Next;
+			Temporal->Next = Current->Next;
+
+			if (Current == Tail)
+			{
+				Tail = Temporal;
+			}
+			delete Current;
+			
+			Size--;
+		}
+	}
+
+	//----------------------------------------------------------
+	void Clear()
+	{
+		FNode* NextItem;
+
+		for (int i = 0; i < Size; i++)
+		{
+			NextItem = Head->Next;
+			delete Head;
+			Head = NextItem;
+		}
+		Size = 0;
+		Head = nullptr;
+		Tail = nullptr;
+	}
+
+	//----------------------------------------ForEach
+	template<typename Function>
+	void forEach(const Function& SomeFunction)
+	{
+		FNode* Temporal = Head;
+		for (int i = 0; i < Size; i++)
+		{
+			SomeFunction(Temporal);
+			Temporal = Temporal->Next;
+		}
+	}
+
+	//----------------------------------------Lambdas
+	template< typename Pred>
+	Type* FindByPredicate(const Pred& Predictate)
```

Comment:
Small typo here, though not too serious as its just on the parameter name. Predicate instead of PredicTate.

Reply:
There's still a typo, it's Predicate

Reply:
Also, const function